language/move-binary-format/src/serializer.rs

```diff
@@ -720,14 +723,32 @@ fn serialize_ability_sets(binary: &mut BinaryData, sets: &[AbilitySet]) -> Resul
 /// - `CodeUnit.max_stack_size` as a ULEB128
 /// - `CodeUnit.locals` as a ULEB128 (index into the `LocalSignaturePool`)
 /// - `CodeUnit.code` as variable size byte stream for the bytecode
-fn serialize_code_unit(binary: &mut BinaryData, code: &CodeUnit) -> Result<()> {
+fn serialize_code_unit(major_version: u32, binary: &mut BinaryData, code: &CodeUnit) -> Result<()> {
     serialize_signature_index(binary, &code.locals)?;
-    serialize_code(binary, &code.code)
+    serialize_code(major_version, binary, &code.code)
 }
 
 /// Serializes a single `Bytecode` instruction.
-fn serialize_instruction_inner(binary: &mut BinaryData, opcode: &Bytecode) -> Result<()> {
+fn serialize_instruction_inner(
+    major_version: u32,
+    binary: &mut BinaryData,
+    opcode: &Bytecode,
+) -> Result<()> {
     let res = match opcode {
+        Bytecode::CastU16
```

Comment:
two nits: first, flip the order of `Cast*` and `Ld*` to match the bytecode enum. second, move these close to the match below

Reply:
Maybe move it out of the match, like we do for vector instructions in the deserializer.